extensions/commands/art/cmd_remote.py

```diff
@@ -0,0 +1,177 @@
+    for r in remotes:
+        if remote_name == r["name"]:
+            raise ConanException(f"Remote '{remote_name}' ({r['url']}) already exist. "
+                                 f"You can remove it using `conan art:remote remove {remote_name}`")
+
+
+def assert_existing_remote(remote_name, remotes):
+    remote_names = [r["name"] for r in remotes]
+    if remote_name not in remote_names:
+        raise ConanException(f"Remote '{remote_name}' does not exist.")
+
+
+def add_default_arguments(subparser):
+    subparser.add_argument("name", help="Name of the remote to add")
+    return subparser
+
+
+@conan_command(group="Custom commands")
+def remote(conan_api: ConanAPI, parser, *args):
+    """
+    Manages Artifactory server and credentials.
+    """
+
+
+@conan_subcommand()
+def remote_add(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Add Artifactory remote and its credentials.
+    """
+    add_default_arguments(subparser)
+    subparser.add_argument("artifactory_url", help="URL of the artifactory server")
+    subparser.add_argument("--user", help="user name for the repository")
+    subparser.add_argument("--password", help="password for the user name")
+
+    args = parser.parse_args(*args)
+
+    artifactory_url = args.artifactory_url.rstrip("/")
+
+    if not args.user or not args.password:
+        raise ConanException(f"User and password are required to authenticate ({artifactory_url}).")
+
+    name = args.name.strip()
+    artifactory_url = args.artifactory_url.rstrip("/")
+    user = args.user.strip()
+    password = args.password.strip()
+
+    remotes = read_remotes()
+    assert_new_remote(name, remotes)
+
+    token = api_request("get", f"{artifactory_url}/api/security/encryptedPassword", user, password)
+    # TODO: manage error with auth
+
+    new_remote = {"name": name,
+                  "url": artifactory_url,
+                  "user": user,
+                  "password": token}
+    remotes.append(new_remote)
+    write_remotes(remotes)
+    cli_out_write(f"Remote '{name}' ({artifactory_url}) added successfully")
+
+
+@conan_subcommand()
+def remote_remove(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    Remove Artifactory remotes.
+    """
+    add_default_arguments(subparser)
+    args = parser.parse_args(*args)
+
+    name = args.name.strip()
+    remotes = read_remotes()
+    assert_existing_remote(name, remotes)
+    artifactory_url = None
+    keep_remotes = []
+    for r in remotes:
+        if name != r["name"]:
+            keep_remotes.append(r)
+        else:
+            artifactory_url = r["url"]
+    write_remotes(keep_remotes)
+    cli_out_write(f"Remote '{name}' ({artifactory_url}) removed successfully")
+
+
+@conan_subcommand()
+def remote_list(conan_api: ConanAPI, parser, subparser, *args):
+    """
+    List Artifactory remotes.
+    """
+    remotes = read_remotes()
+    if remotes:
+        for r in remotes:
+            cli_out_write(f"{r['name']}:")
```

Comment:
Usually cli_out_write is used for command results and `ConanOutput()` for this kind of logs. It's a bit of bikeshedding for a extension command, but figured I'd mention in case you want to switch :)